### DISCUSSION TITLE
React Query and Next SSG

### DISCUSSION THREAD

**Discussion by esharmony** [TanStack/query#2050] [Q&A] [answered]
Am i right in thinking that there is not much benefit to using React Query when creating a SSG Next.JS site.
When static sites are created there is not much real state to be managed, so all the benefits of react query, caching etc is not used as the data is only requested and used for the pages to be built once on export.

Am I missing anything as I am just at the point of ripping out many of my hooks.

My implantation if of interest can be found at https://github.com/esharmony/personal-blog-nextjs, it was my first attempt at using react query so be kind : ) seriously though always happy for pointers information and help.

**Comment by tannerlinsley** [comment, marked as answer]
As long as you do not need any of the data on the site to update during the user's session (which *would* require some kind of runtime querying), you do not need React Query :)

**Reply by esharmony** [reply]
Many thanks for your response.
